pymodbus/datastore/store.py

```diff
@@ -263,45 +266,50 @@ def getValues(self, address, count=1):
         """
         return [self.values[i] for i in range(address, address + count)]
 
-    def _process_values(self, values):
+    def _process_values(self, values: Union[Mapping, List]):
         """Process values."""
 
-        def _process_as_dict(values):
+        def _process_as_mapping(values):
             for idx, val in iter(values.items()):
                 if isinstance(val, (list, tuple)):
                     for i, v_item in enumerate(val):
                         self.values[idx + i] = v_item
                 else:
                     self.values[idx] = int(val)
 
-        if isinstance(values, dict):
-            _process_as_dict(values)
+        if isinstance(values, abc.Mapping):
+            _process_as_mapping(values)
             return
         if hasattr(values, "__iter__"):
             values = dict(enumerate(values))
         elif values is None:
             values = {}  # Must make a new dict here per instance
         else:
             raise ParameterException(
-                "Values for datastore must be a list or dictionary"
+                "Values for datastore must be a mapping or dictionary"
             )
-        _process_as_dict(values)
-
-    def setValues(self, address, values, use_as_default=False):
+        _process_as_mapping(values)
+
+    def setValues(
+        self,
+        address: int,
+        values: Union[Mapping, Collection, float],
```

Comment:
setValues only uses an int or list[int].

Please when you make such rather deep changes, have a look at where it is used.

Reply:
Of course I look at usage.  That's why I had to expand the typing of `ModbusSparseDataBlock._process_values()` when I changed `setValues()`

Reply:
Well then you would have seen that get/set/validate operate on registers which are lists of UINT16 and nothing else.

You have 3 angels on datastore:
- the definition or setup which is from outside and need to flexible.
- the usage which is from pymodbus internal only, and is only using list[int]
- the internal storage, which are different in the different types of datastore

Each view have its own set of rules, please do not try to make those 3 identical, that is wrong.